### PR TITLE
Added test flow for creating, getting, parsing and deleting a Consul …

### PR DESCRIPTION
**Pull request by tudorlesan**
…key.

Signed-off-by: Lesan Tudor <tudor-andrei.lesan@hp.com>

### CODE REVIEW

**Comment by tudorlesan** [comment]
test_report_machine_metrics_cAdvisor fails sporadically but consul tests pass.

**Comment by meirwah** [comment]
seems hat test_report_machine_metrics_cAdvisor  keeps failing here.. regression?

**Comment by orius123** [comment]
Looks like it is. But Y??

**Comment by meirwah** [comment]
I do not see in the changes something that can effect it...

**Comment by meirwah** [comment]
????

**Comment by orius123** [comment]
Only thing that I can think of is disk space being full.

**Comment by meirwah** [comment]
why did you close this??

**Comment by orius123** [comment]
Oops

**Comment by meirwah** [comment]
might be.. is there a way to see if it is disk space related??

**Comment by orius123** [comment]
Connect with ssh to the machine after the build is done. Or look in the builder.log

**Comment by meirwah** [comment]
are you exporting the builder.log??

**Comment by meirwah** [comment]
https://circle-artifacts.com/gh/CloudSlang/cloud-slang-content/917/artifacts/0/home/ubuntu/cloud-slang-content/builder.log

**Comment by orius123** [comment]
Yup. This how awesome we are :smile:

**Comment by meirwah** [comment]
you know what this means:
Parse error on line 1: [ INFO] [ INFO] -----^ Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', ']'

**Comment by orius123** [comment]
I think trying to merge from #218 should solve it.

**Comment by orius123** [comment]
regarding the parse error. I have no idea..

**Comment by meirwah** [comment]
@tudorlesan can you take a look at the logs?

**Comment by tudorlesan** [comment]
@meirwah I will look into them, though I think that @orius123 might be right. The test might fail because the container does not get the chance to get up fully and the next steps of the flow fail. So a sleep would solve the problem. But I'll check

**Comment by tudorlesan** [comment]
@meirwah this is the error message: errorMessage=Error while executing http request: localhost:32951 failed to respond

I really think that merging #218 will solve this problem.

**Comment by meirwah** [comment]
merged #218 , lets see...

**Comment by tudorlesan** [comment]
:+1:

**Comment by meirwah** [comment]
seems it works now...

**Comment by meirwah** [comment]
@Bonczidai please review this...

File: content/io/cloudslang/consul/get_catalog_services.sl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+# Gets a list of services in a given datacenter.
+#
+# Inputs:
+#   - host - Consul agent host
+#   - consul_port - optional - Consul agent port - Default: 8500
+#   - datacenter - optional - Default: matched to that of agent
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/get_catalog_services.sl]: missing actual description, datacenter is declared required without default value in the inputs section

File: content/io/cloudslang/consul/get_catalog_services.sl [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+operation:
+  name: get_catalog_services
+  inputs:
+    - host
+    - consul_port:
+        default: "'8500'"
+        required: false
+    - datacenter:
+        required: false
+    - dc: "'?dc=' + datacenter if bool(datacenter) else ''"
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/get_catalog_services.sl]: overridable: false?

File: content/io/cloudslang/consul/parse_register_agent_service_request.sl [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+# Creates JSON for request to register new agent service.
+#
+# Inputs:
+#   - address - optional - will default to that of the agent
+#   - service - optional - if Service key is provided, then service will also be registered
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/parse_register_agent_service_request.sl]: no `service` input but `service_name` and `service_id`

File: content/io/cloudslang/consul/parse_register_agent_service_request.sl [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+####################################################
+
+namespace: io.cloudslang.consul
+
+operation:
+  name: parse_register_agent_service_request
+  inputs:
+    - address:
+        default: "''"
+        required: false
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/parse_register_agent_service_request.sl]: couldn't you omit the empty strings by check in the script if it exists? `bool(input)`

File: content/io/cloudslang/consul/parse_register_agent_service_request.sl [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          data['ID'] = service_id
+        data['Name'] = service_name
+        if check != '':
+          data['Check'] = json.loads(check)
+        json_request = json.dumps(data)
+        returnCode = '0'
+        returnResult = 'Parsing successful.'
+      except:
+        returnCode = '-1'
+        returnResult = 'Parsing error'
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/parse_register_agent_service_request.sl]: maybe you could include the error message in the returnResult like `returnResult = 'Parsing error: ' + str(ex)`

File: content/io/cloudslang/consul/register_agent_service.sl [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+#
+# Inputs:
+#   - host - Consul agent host
+#   - consul_port - optional - Consul agent host port - Default: 8500
+#   - address - optional - will default to that of the agent
+#   - service_name - the service name what will be registered
+#   - service_id - optional - if omitted, service_name will be used instead
+#   - check - optional - if the Check key is provided, then a health check will also be registered
+# Outputs:
+#   - errorMessage - returnResult if there was an error
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/register_agent_service.sl]: missing returnResult output

File: content/io/cloudslang/consul/register_agent_service.sl [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+              - json_request
+        publish:
+            - errorMessage
+            - returnResult
+  outputs:
+      - errorMessage
+      - returnResult
+  results:
+      - SUCCESS
+      - FAILURE
```
Comment (Bonczidai) [inline content/io/cloudslang/consul/register_agent_service.sl]: should not declare default results

**Comment by Bonczidai** [comment]
added my comments

File: content/io/cloudslang/consul/get_catalog_services.sl (hunk 1 above)
Comment (tudorlesan) [inline content/io/cloudslang/consul/get_catalog_services.sl]: the default value is given by the consul API not by the flow itself. That's why it does not contain a default value in the flow, but contains default in documentation

**Pull request merged**